src/nunavut/lang/__init__.py

```diff
@@ -41,22 +57,50 @@ def get_module(self) -> typing.Any:
             self._module = importlib.import_module('nunavut.lang.{}'.format(self._language_name))
         return self._module
 
+    def get_templates_package_name(self) -> str:
+        return 'nunavut.lang.{}'.format(self.name)
+
 
 class LanguageContext:
     """
     Context object containing the current target language (if any) and used to access
     :class:`Language` objects.
+
+    :param str target_language: If provided a :class:`Language` object will be created to hold
+                                the target language set for this context. If None then there is
+                                no target language.
+    :param str extension: The extension to use for generated file types or None to use a default
+                          based on the target_language.
+    :param str namespace_output_stem: The filename stem to give to Namespace output files if
+                                      emitted or None to use a default based on a target_language.
+    :raises ValueError: If extension is None and no target language was provided.
+    :raises KeyError: If the target language is not known.
     """
 
-    _default_output_extensions = {
-        'c': '.h',
-        'cpp': '.hpp',
-        'js': '.js',
-        'py': '.py'
+    _language_defaults = {
+        'c': ('.h', '__module__'),
+        'cpp': ('.hpp', '__module__'),
```

Comment:
Why double underscores? Would `_module_` not be sufficient, considering that `_.*_` is a reserved pattern in DSDL? Double underscores scare me because they are *so long*, whenever you encounter one it just feels like the underscore is going to extend into infinity and never end, your life and whole existence would just become one epic underscore that can only be ended by the heat death of the universe :scream_cat:

Reply:
Wow. That's terrifying. I'd better fix this.

Reply:
Hmm. __init__ is mandated by Python so that's a required choice. What about the rest. Any suggestions for what we should call C, C++, and JavaScript source that defines a DSDL namespace?